### PR TITLE
CF-200 Optimized imports in testcases and test classes.

### PR DESCRIPTION
**Pull request by plomakin**
Added import optimization for test classes.
Moved generate_load to devlab dir.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2546/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.generate_load
devlab/generate_load.py:587: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/generate_load.py:661: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
************* Module devlab.tests.utils
devlab/tests/utils.py:34: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:135: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module devlab.tests.testcases.test_resource_migration
devlab/tests/testcases/test_resource_migration.py:670: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/testcases/test_resource_migration.py:680: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.generate_load
devlab/generate_load.py:587: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/generate_load.py:661: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
************* Module devlab.tests.utils
devlab/tests/utils.py:135: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2559/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2562/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.generate_load
devlab/generate_load.py:665: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.generate_load
devlab/generate_load.py:665: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/generate_load.py [hunk 1]
```diff
@@ -12,23 +12,24 @@
...
 from neutronclient.common import exceptions as nt_exceptions
 from novaclient import exceptions as nv_exceptions
 
-from base import BasePrerequisites
-from cleanup import CleanEnv
-import config as conf
-import logging
-from logging.config import dictConfig
+import devlab.tests.config as conf
+from devlab.tests.base import BasePrerequisites
```
Comment (vladiskuz) [inline devlab/generate_load.py]: I think that would be better to import module, not class. (smth like "from devlab.tests import base')

File: devlab/generate_load.py [hunk 2]
```diff
@@ -12,23 +12,24 @@
...
+from devlab.tests.base import get_dict_from_config_file, get_abs_path
+from devlab.tests.cleanup import CleanEnv
```
Comment (vladiskuz) [inline devlab/generate_load.py]: The same thing for the cleanup module.

**Comment by vladiskuz** [comment]
I have concern about importing classes and functions from module. By my opinion we should import module, not class. I think that will be better if we will use module_name.Class_name or module_name.function_name in code - it will be more readable.

**Comment by vladiskuz** [comment]
Please, also fix documentation. This patch will move generate_load.py script, because of this we should change path to this script at least in these three places:

- https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/docs/CloudFerry_Beginner_Demo_How-To.md
- https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/QUICKSTART.md
- on the wiki page

Also generate_load.py used here:
- https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/devlab/utils/clean_run.sh
- https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/devlab/utils/deploy_lab.sh

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: .pylintrc [hunk 3]
```diff
@@ -7,7 +7,7 @@
 # W0622: Redefining id is fine.
 # C: Disable all conventional warnings
 # R: Disable all refactoring warnings
-disable=I0011,W0511,W0142,W0622,C,R
+disable=I0011,W0511,W0141,W0142,W0622,C,R
```
Comment (roman-verchikov) [inline .pylintrc]: Let's keep W0141 warnings, it leads to more concise and standardized code

File: devlab/generate_load.py [hunk 4]
```diff
@@ -669,7 +662,7 @@ def get_params_for_volume_creating(_volume):
                                  tenant=user['tenant'])
             vlm = self.cinderclient.volumes.create(
                 **get_params_for_volume_creating(volume))
-            vlm_ids.append(vlm.id)
+            vlm_ids.append(vlm.__dict__.get('id'))
```
Comment (roman-verchikov) [inline devlab/generate_load.py]: Looks terrible... Why would `vlm.id` not work?..

File: devlab/generate_load.py [hunk 5]
```diff
@@ -597,7 +590,7 @@ def create_security_group(self, sg_list):
         for security_group in sg_list:
             gid = self.novaclient.security_groups.create(
                 name=security_group['name'],
-                description=security_group['description']).id
+                description=security_group['description']).to_dict().get('id')
```
Comment (roman-verchikov) [inline devlab/generate_load.py]: maybe 
```
sg = novaclient.create(...)
...
sg.id
```
?

File: devlab/tests/testcases/test_resource_migration.py [hunk 6]
```diff
@@ -372,7 +372,7 @@ def test_glance_images_not_in_filter_did_not_migrate(self):
                             'in filter, Image info: \n{}'.format(image))
 
     def test_migrate_neutron_networks(self):
-        """Validate networks were migrated with correct parameters.
+        r"""Validate networks were migrated with correct parameters.
```
Comment (roman-verchikov) [inline devlab/tests/testcases/test_resource_migration.py]: Hmm... looks strange. Maybe the actual problem is in backslashes in the text?

**Comment by roman-verchikov** [comment]
Please resolve inlines

File: devlab/generate_load.py (hunk 4 above)
Comment (plomakin) [inline devlab/generate_load.py]: it works, but pylint shows "Instance of 'dict' has no 'id' member"

File: devlab/generate_load.py (hunk 5 above)
Comment (plomakin) [inline devlab/generate_load.py]: in this case pylint shows "Instance of 'dict' has no 'id' member"

File: .pylintrc (hunk 3 above)
Comment (plomakin) [inline .pylintrc]: ok

File: devlab/tests/testcases/test_resource_migration.py (hunk 6 above)
Comment (plomakin) [inline devlab/tests/testcases/test_resource_migration.py]: There are two ways to resolve this pylint issue - will try another one in next iteration

File: devlab/generate_load.py (hunk 5 above)
Comment (roman-verchikov) [inline devlab/generate_load.py]: I guess this is the case where we should simply ignore pylint issue

File: devlab/generate_load.py (hunk 4 above)
Comment (roman-verchikov) [inline devlab/generate_load.py]: Let's better ignore pylint issue, `vlm.__dict__.get('id')` is not worth it

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.tests.utils
devlab/tests/utils.py:34: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
************* Module devlab.generate_load
devlab/generate_load.py:591: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/generate_load.py:665: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

File: QUICKSTART.md [hunk 7]
```diff
@@ -66,10 +66,10 @@
 
 7. Generate load on source VM (this will create a number of VMs on grizzly node)
     ```
-    cd CloudFerry/devlab/tests
+    cd CloudFerry/
```
Comment (roman-verchikov) [inline QUICKSTART.md]: `cd CloudFerry/devlab/tests`

File: devlab/tests/utils.py [hunk 8]
```diff
@@ -132,7 +132,7 @@ def execute_command_on_vm(self, ip_addr, cmd, username=None,
             username = self.config.username_for_ssh
         if password is None and key is None:
             password = self.config.password_for_ssh
-        with hide('everything'), settings(
+        with hide('everything') and settings(
```
Comment (roman-verchikov) [inline devlab/tests/utils.py]: `settings(hide('everything'), ...)`

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module devlab/tests/generate_load.py
devlab/tests/generate_load.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load.py
************* Module devlab/tests/generate_load_real_env.py
devlab/tests/generate_load_real_env.py:1: [F0001(fatal), ] No module named devlab/tests/generate_load_real_env.py
************* Module devlab/tests/real_env_conf.py
devlab/tests/real_env_conf.py:1: [F0001(fatal), ] No module named devlab/tests/real_env_conf.py
************* Module devlab.tests.utils
devlab/tests/utils.py:34: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
************* Module devlab.generate_load
devlab/generate_load.py:591: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/generate_load.py:665: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Pull request merged**